infrastructure/code/code_test.go

```diff
@@ -61,176 +62,237 @@ func setupDocs() (string, lsp.TextDocumentItem, lsp.TextDocumentItem, []byte, []
+			}
+			assert.Len(t, missingFiles, 1, "bundle should have 1 missing files")
+			assert.Len(t, snykCodeMock.GetAllCalls(CreateBundleOperation), 1, "bundle should called createBundle once")
+		},
+	)
+
+	t.Run(
+		"when too big ignores file", func(t *testing.T) {
+			snykCodeMock, dir, c, file := setupCreateBundleTest(t, "java")
+			data := strings.Repeat("a", maxFileSize+1)
+			err := os.WriteFile(file, []byte(data), 0600)
+			if err != nil {
+				t.Fatal(err)
+			}
+			_, missingFiles, err := c.createBundle(context.Background(), "testRequestId", dir, []string{file})
+			if err != nil {
+				t.Fatal(err)
+			}
+			assert.Len(t, missingFiles, 0, "bundle should not have missing files")
+			assert.Len(t, snykCodeMock.GetAllCalls(CreateBundleOperation), 0, "bundle shouldn't have called createBundle")
+		},
+	)
+
+	t.Run(
+		"when empty file ignores file", func(t *testing.T) {
+			snykCodeMock, dir, c, file := setupCreateBundleTest(t, "java")
+			fd, err := os.Create(file)
+			t.Cleanup(
+				func() {
+					_ = fd.Close()
+				},
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+			_, missingFiles, err := c.createBundle(context.Background(), "testRequestId", dir, []string{file})
+			if err != nil {
+				t.Fatal(err)
+			}
+			assert.Len(t, missingFiles, 0, "bundle should not have missing files")
+			assert.Len(t, snykCodeMock.GetAllCalls(CreateBundleOperation), 0, "bundle shouldn't have called createBundle")
+		},
+	)
+
+	t.Run(
+		"when unsupported ignores file", func(t *testing.T) {
+			snykCodeMock, dir, c, file := setupCreateBundleTest(t, "unsupported")
+			fd, err := os.Create(file)
+			t.Cleanup(
+				func() {
+					_ = fd.Close()
+				},
+			)
+			if err != nil {
+				t.Fatal(err)
+			}
+			_, missingFiles, err := c.createBundle(context.Background(), "testRequestId", dir, []string{file})
+			if err != nil {
+				t.Fatal(err)
+			}
+			assert.Len(t, missingFiles, 0, "bundle should not have missing files")
+			assert.Len(t, snykCodeMock.GetAllCalls(CreateBundleOperation), 0, "bundle shouldn't have called createBundle")
+		},
+	)
 }
 
 func setupCreateBundleTest(t *testing.T, extension string) (*FakeSnykCodeClient, string, *Scanner, string) {
 	testutil.UnitTest(t)
 	dir := t.TempDir()
-	snykCodeMock, c := setupTestScanner()
+	snykCodeMock, c := setupTestScannerWithNotifications()
 	file := filepath.Join(dir, "file."+extension)
 	return snykCodeMock, dir, c, file
 }
 
-func setupTestScanner() (*FakeSnykCodeClient, *Scanner) {
+func setupTestScannerWithNotifications() (*FakeSnykCodeClient, *Scanner) {
 	snykCodeMock := &FakeSnykCodeClient{}
 	scanner := New(
 		NewBundler(snykCodeMock, performance.NewTestInstrumentor()),
 		&snyk_api.FakeApiClient{CodeEnabled: true},
 		error_reporting.NewTestErrorReporter(),
 		ux2.NewTestAnalytics(),
+		notification.NewScanNotifier(notification.NewNotifier(), "code"),
 	)
 
 	return snykCodeMock, scanner
 }
 
+func setupTestScanner() (*FakeSnykCodeClient, *notification.MockScanNotifier, *Scanner) {
```

Comment:
Why having two functions, when you can parametrise the above one to use mocknotifier?

Reply:
The original `setupTestScanner` is used for tests that assert results by actively listening to the messages in `notification`, so I'd have to refactor many tests to do it. Adding a parameter also requires to refactor all the calls to the setup function.